src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionStringBuilder.cs

```diff
@@ -447,8 +452,9 @@ protected virtual void GetProperties(Hashtable propertyDescriptors)
                 if (!IsFixedSize)
                 {
                     attributes = null;
-                    foreach (string keyword in Keys)
+                    foreach (string? keyword in Keys)
```

Comment:
Ditto; in general now these `?` when foreaching through non-generic `IEnumerable`s shouldn't be needed. (I'm assuming Keys is non-generic, but haven't actually looked)

Reply:
Yeah, it's all non-generic. But isn't it always possible to have nulls in the non-generic collections? Or is this a way of asserting that no nulls come out of the collection?

Reply:
> But isn't it always possible to have nulls in the non-generic collections?

It is.  But because:
a) you can `foreach` a non-generic collection and put any type you want as the iteration type without warning
b) it's rare to use the non-generic `IEnumerator.Current` directly, and
c) we have lots of collections that implement the non-generic interface but won't contain nulls
we recently decided to make `IEnumerator.Current` oblivious.

Reply:
OK, thanks for explaining - will adjust. Unfortunately there's lots of other access of non-generic collections (e.g. via indexers) which don't contain null, and which I currently must suppress. We may want to think about replacing some with their generic counterparts, although on the other hand it's probably better to leave most of this code alone.